WordPressAuthenticator/Unified Auth/View Related/Reusable Views/TextFieldTableViewCell.swift

```diff
@@ -52,23 +61,96 @@ private extension TextFieldTableViewCell {
     func setCommonTextFieldStyles() {
         textField.font = UIFont.preferredFont(forTextStyle: .body)
         textField.autocorrectionType = .no
-        textField.returnKeyType = .continue
     }
 
     /// Sets the textfield keyboard type and applies common traits.
     /// - note: Don't assign first responder here. It's too early in the view lifecycle.
     ///
     func applyTextFieldStyle(_ style: TextFieldStyle) {
-        switch style {
+		switch style {
         case .url:
             textField.keyboardType = .URL
-        default:
-            setCommonTextFieldStyles()
+			textField.returnKeyType = .continue
+		case .username:
+			textField.keyboardType = .default
+			textField.returnKeyType = .next
+		case .password:
+			textField.keyboardType = .default
+			textField.returnKeyType = .continue
+			setSecureTextEntry(true)
+			showSecureTextEntryToggle = true
+			configureSecureTextEntryToggle()
         }
     }
 }
 
 
+// MARK: - Secure Text Entry
+/// Methods ported from WPWalkthroughTextField.h/.m
+///
+private extension TextFieldTableViewCell {
+
+	/// Build the show / hide icon in the textfield.
+	///
+	func configureSecureTextEntryToggle() {
+		if showSecureTextEntryToggle == false {
+			return
+		}
+
+		secureTextEntryImageVisible = UIImage.gridicon(.visible)
+		secureTextEntryImageHidden = UIImage.gridicon(.notVisible)
+
+		secureTextEntryToggle = UIButton(type: .custom)
+		secureTextEntryToggle?.clipsToBounds = true
+		// The icon should match the border color.
+		let tintColor = WordPressAuthenticator.shared.unifiedStyle?.borderColor ?? WordPressAuthenticator.shared.style.primaryNormalBorderColor
+		secureTextEntryToggle?.tintColor = tintColor
+
+		secureTextEntryToggle?.addTarget(self,
+										 action: #selector(secureTextEntryToggleAction),
+										 for: .touchUpInside)
+
+		updateSecureTextEntryToggleImage()
+		updateSecureTextEntryForAccessibility()
+		textField.rightView = secureTextEntryToggle
+		textField.rightViewMode = .always
+	}
+
+	func setSecureTextEntry(_ secureTextEntry: Bool) {
+		// This is a fix for a bug where the text field reverts to a system
+		// serif font if you disable secure text entry while it contains text.
+		textField.font = nil
```

Comment:
Is setting it to `nil` necessary since it's immediately followed by setting the font? I commented it out and didn't see a difference.